### PR TITLE
:bug: reconciler/apibinding: share conflict checker for all schemas in bindings

### PR DESCRIPTION
**Pull request by sttts**
## Summary

Share the conflict checker when checking all resources in a APIBinding. The checker showed up pprof because it does don't of informer lookups. With https://github.com/kcp-dev/kcp/pull/3169, this should be a lot better.

## Related issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

### CODE REVIEW

File: pkg/reconciler/apis/apibinding/apibinding_reconcile.go [hunk 1]
```diff
@@ -207,11 +207,15 @@ func (r *bindingReconciler) reconcile(ctx context.Context, apiBinding *apisv1alp
 
 	// Record the APIExport's host cluster name for lookup in webhooks.
 	// The full path is unreliable for this purpose.
-	clusterName := logicalcluster.From(apiExport)
-	apiBinding.Status.APIExportClusterName = clusterName.String()
+	apiBinding.Status.APIExportClusterName = logicalcluster.From(apiExport).String()
 
 	var needToWaitForRequeueWhenEstablished []string
 
+	checker, err := NewConflictChecker(logicalcluster.From(apiBinding), r.listAPIBindings, r.getAPIExportByPath, r.getAPIResourceSchema, r.getCRD, r.listCRDs)
```
Comment (mjudeikis) [inline pkg/reconciler/apis/apibinding/apibinding_reconcile.go:214]: Any reason this is exported? This looks like internal helpers inside reconcile loop.

**Comment by mjudeikis** [comment]
/lgtm
/approve
/hold
not blocker so we can merge if needed

File: pkg/reconciler/apis/apibinding/apibinding_reconcile.go (hunk 1 above)
Comment (sttts) [reply pkg/reconciler/apis/apibinding/apibinding_reconcile.go]: fixed

**Pull request merged**